petab/problem.py

```diff
@@ -109,6 +115,41 @@ def from_files(sbml_file: str = None,
                        sbml_document=sbml_document,
                        sbml_reader=sbml_reader)
 
+    @staticmethod
+    def from_yaml(yaml_config: Union[Dict, str]) -> 'Problem':
+        """
+        Factory method to load model and tables as specified by YAML file.
+
+        Arguments:
+            yaml_config: PEtab configuration as dictionary or YAML file name
+        """
+        old_wd = os.getcwd()
+        if isinstance(yaml_config, str):
+            new_wd = os.path.dirname(yaml_config)
+            yaml_config = yaml.load_yaml(yaml_config)
+
+        if yaml.is_composite_problem(yaml_config):
+            raise ValueError('petab.Problem.from_yaml() can only be used for '
+                             'yaml files comprising a single model. '
+                             'Consider using '
+                             'petab.CompositeProblem.from_yaml() instead.')
+        try:
```

Comment:
for future reference, this repeating code pattern might be nice to address with a context manager.

Reply:
Right. But now removed anyways...